lib/luahid.c

```diff
@@ -93,6 +93,66 @@ static const struct hid_device_id *luahid_setidtable(lua_State *L, int idx)
 	return user_table;
 }
 
+#define luahid_setfield(L, idx, obj, field)	\
+do { 						\
+	lua_pushinteger(L, obj->field);		\
+	lua_setfield(L, idx - 1, #field);	\
+} while (0)
+
+static inline void luahid_pushdevid(lua_State *L, int idx, const struct hid_device_id *device_id)
+{
+	lua_newtable(L);
+	luahid_setfield(L, -1, device_id, bus);
+	luahid_setfield(L, -1, device_id, group);
+	luahid_setfield(L, -1, device_id, vendor);
+	luahid_setfield(L, -1, device_id, product);
+	luahid_setfield(L, -1, device_id, driver_data);
+}
+
+#define luahid_checkhandlertable(L, hid, idx, field) (lunatik_getregistry(L, hid) != LUA_TTABLE || \
+	lua_getfield(L, idx, "ops") != LUA_TTABLE || lua_getfield(L, idx - 1, field) != LUA_TTABLE)
+
+static int luahid_probe_handler(lua_State *L, luahid_t *hid, struct hid_device *hdev, const struct hid_device_id *id)
```

Comment:
I would apply the same nomenclature we are using on luadevice..
```suggestion
static int luahid_doprober(lua_State *L, luahid_t *hid, struct hid_device *hdev, const struct hid_device_id *id)
```

Reply:
done